packages/plugins/src/foundry/utils/trim.ts

```diff
@@ -0,0 +1,143 @@
+import { BuildInfo } from '@sphinx-labs/core'
+
+export const trimObjectToType = <T>(obj: any, typeTemplate: T): T => {
+  // If the template is either an array then we just return since we don't do any filtering on arrays
+  if (Array.isArray(typeTemplate)) {
+    return obj
```

Comment:
If i'm understanding this correctly, it seems this function doesn't trim any elements that exist inside of an array. Is that right? If so, we may want to either fix that or just add docs to this function mentioning this limitation